### PR TITLE
fix storage module sync on active tab

### PR DESCRIPTION
**Pull request by unadlib**

### CODE REVIEW

**Comment by u9520107** [comment]
I don't think it is safe to disable storage sync based on tabmanager. Consider the case when tab1 has started to request for permissions, and tab2 skipped request because it is not the active tab. If user sit he'd to tab2 before request was completed, then tab1 will not be able to sync the requested data when it completes. Tab2 will most likely wait through the default retry interval to start requesting for permissions. You'll have drastically different states across tabs. I think carefully adding tabmanager control to things that use storage modules is better albeit more tedious.

**Comment by embbnux** [comment]
@u9520107 In this PR, it just disables to sync from redux data to localstorage data. Tab2 in inactive tab can still sync data from localstorage to redux data.   This is to fix bug that tab1 in active tab will sync wrong data from tab2 in inactive tab when tab2 save data to localstorage data.

File: packages/ringcentral-integration/modules/Storage/index.js [hunk 1]
```diff
@@ -81,7 +87,10 @@ export default class Storage extends StorageBase {
           type: this.actionTypes.resetSuccess,
         });
       }
-      if (this.status !== moduleStatuses.pending) {
+      if (
+        this.status !== moduleStatuses.pending &&
```
Comment (u9520107) [inline packages/ringcentral-integration/modules/Storage/index.js]: I'd suggest to only sync on this.status === moduleStatuses.ready instead.
Even though it's equivalent to the current behavior, but I think explicitly use ready state is more straight forward.

Comment (unadlib) [reply packages/ringcentral-integration/modules/Storage/index.js]: OK, I also feel so good.

File: packages/ringcentral-integration/modules/Storage/index.js [hunk 2]
```diff
@@ -3,29 +3,41 @@ import StorageBase from '../../lib/StorageBase';
 import loginStatus from '../Auth/loginStatus';
 import moduleStatuses from '../../enums/moduleStatuses';
 
+const DEFAULT_ALLOW_INACTIVE_TABS_WRITE = false;
```
Comment (embbnux) [inline packages/ringcentral-integration/modules/Storage/index.js]: It will be better to allow write by default. So we doesn't change default behavior.

**Pull request merged**